SU2_CFD/include/solver_structure.hpp

```diff
@@ -4684,6 +4682,8 @@ class CEulerSolver : public CSolver {
 
   CFluidModel  *FluidModel;  /*!< \brief fluid model used in the solver */
 
+  CEulerVariable* snode;  /*!< \brief The highest level in the variable hierarchy this solver can safely use. */
```

Comment:
As I mentioned in #716 the solver methods that access primitives (gradient and limiter computation) suffer from the accessors to those variables being virtual (therefore not inline-able if called on pointer to base class). To fix that, in some solvers I have defined a `snode` variable of the type that solver can safely assume will be instantiated.